Sources/ApolloWebSocket/WebSocketTransport.swift

```diff
@@ -413,35 +442,23 @@ extension WebSocketTransport: WebSocketDelegate {
       self.delegate?.webSocketTransportDidConnect(self)
     }
 
-    reconnected = true
+    self.reconnected = true
   }
 
-  public func websocketDidDisconnect(socket: WebSocketClient, error: Error?) {
-    // report any error to all subscribers
-    if let error = error {
-      self.error.mutate { $0 = WebSocketError(payload: nil,
-                                              error: error,
-                                              kind: .networkError) }
-      self.notifyErrorAllHandlers(error)
-    } else {
-      self.error.mutate { $0 = nil }
-    }
-
+  private func handleDisconnection()  {
     self.delegate?.webSocketTransport(self, didDisconnectWithError: self.error.value)
-    acked = false // need new connect and ack before sending
+    self.acked = false // need new connect and ack before sending
 
-    if reconnect.value {
-      DispatchQueue.main.asyncAfter(deadline: .now() + reconnectionInterval) {
-        self.websocket.connect()
-      }
-    }
+    self.attemptReconnectionIfDesired()
   }
-
-  public func websocketDidReceiveMessage(socket: WebSocketClient, text: String) {
-    processMessage(socket: socket, text: text)
-  }
-
-  public func websocketDidReceiveData(socket: WebSocketClient, data: Data) {
-    processMessage(socket: socket, data: data)
+  
+  private func attemptReconnectionIfDesired() {
+    guard self.reconnect.value else {
+      return
+    }
+    
+    DispatchQueue.main.asyncAfter(deadline: .now() + reconnectionInterval) {
```

Comment:
Is it right to always be doing this on the main queue? Do we have to use main for some reason here?

Reply:
That's a reasonable question - this bit was basically moving around some existing code. 

I suspect part of the reason it's on `main` is because we do the initial connection on `main` by default, but I suspect the bigger reason is because you can't just grab the current queue and dispatch on it the way you could in ObjC.